Fix OOM test for HOST ALL memspace

kswiecicki opened the pull request:
Adjust alloc size to 4MB. Add an access to each allocation so that pages are actually allocated on the selected NUMA nodes.

~~Fixes: https://github.com/oneapi-src/unified-memory-framework/issues/285~~

This also fixes the nightly workflow fails here:  https://github.com/oneapi-src/unified-memory-framework/actions/runs/8148821384.
Nightly workflow run with these changes on my fork: https://github.com/kswiecicki/unified-memory-framework/actions/runs/8188227540.

ldorau commented:
@kswiecicki CI build fail

lukaszstolarczuk commented:
@kswiecicki, I can see you're doing some changes yet - is it ready to review/merge at this moment? 😉  if you still testing this, consider changing this PR to draft, please.

ldorau commented:
The Nightly build with your PR fails: https://github.com/ldorau/unified-memory-framework/actions/runs/8169938615/job/22335017162

kswiecicki commented:
> The Nightly build with your PR fails: https://github.com/ldorau/unified-memory-framework/actions/runs/8169938615/job/22335017162

After talking to @igchor and discovered that `provider_os_memory_config` tests had similar problems with memory policy assertions under valgrind. For that reason I've added a similar suppression for `memspace_host_all` test. Nightly tests shouldn't fail now.

kswiecicki commented:
I've rewritten the test case according to what we came up with on standup @bratpiorka. It seems that OOM killer no longer kills it. Also, I've added a skip for this test case if it was built with TSan, because it exits with `143` error code without providing any further info.

ldorau reviewed (approved):
Valgrind memcheck build PASSes: https://github.com/ldorau/unified-memory-framework/actions/runs/8188719802
